src/types/Profiles.cpp

```diff
@@ -1,13 +1,18 @@
 #include "../../include/types/Profiles.h"
+#include "../../include/types/Perf.h"
+#include "../../include/types/Performance.h"
 #include "../../include/types/PerformanceTable.h"
 #include "../../include/utils.h"
 #include <algorithm>
 #include <iostream>
+#include <random>
+#include <sstream>
 #include <string>
 #include <vector>
 
-Profiles::Profiles(std::vector<Performance> &perf_vect)
+Profiles::Profiles(std::vector<Performance> &perf_vect, std::string mode)
     : PerformanceTable(perf_vect) {
+  mode_ = mode;
```

Comment:
Maybe we should have a check for the mode here (for example if the given mode is not a valid argument)